buildSrc/src/test/groovy/org/elasticsearch/gradle/BuildPluginTests.java

```diff
@@ -36,4 +39,20 @@ public void testFailingDockerVersions() {
         BuildPlugin.checkDockerVersionRecent("Docker version 17.04.0, build e68fc7a");
     }
 
+    @Test(expected = GradleException.class)
+    public void testRepositoryURIThatUsesHttpScheme() throws URISyntaxException {
+        final URI uri = new URI("http://s3.amazonaws.com/artifacts.elastic.co/maven");
+        BuildPlugin.assertRepositoryURIIsSecure("test", "test", uri);
+    }
+
+    public void testRepositoryURIThatUsesHttpsScheme() throws URISyntaxException {
```

Comment:
Can we add a test for `file:` as well?

Reply:
I pushed f028816b2cf3e7f37e9e4be77e05fd9d94372116.